src/Symfony/Component/ObjectMapper/ObjectMapper.php

```diff
@@ -237,7 +228,16 @@ private function getSourceValue(object $source, object $target, mixed $value, \S
             } elseif ($objectMap->offsetExists($value)) {
                 $value = $objectMap[$value];
             } else {
-                $value = ($this->objectMapper ?? $this)->map($value, $mapTo->target);
+                if (\PHP_VERSION_ID >= 80400) {
+                    $refl = new \ReflectionClass($mapTo->target);
+                    $mapper = $this->objectMapper ?? $this;
+
+                    return $refl->newLazyProxy(static function () use ($mapper, $value, $objectMap, $mapTo) {
+                        return $objectMap[$value] = $mapper->map($value, $mapTo->target);
+                    });
+                }
+
+                return ($this->objectMapper ?? $this)->map($value, $mapTo->target);
```

Comment:
to minimize the diff with 8.0:
```suggestion
            } elseif (\PHP_VERSION_ID < 80400) {
                return ($this->objectMapper ?? $this)->map($value, $mapTo->target);
            } else {
                $refl = new \ReflectionClass($mapTo->target);
                $mapper = $this->objectMapper ?? $this;

                return $refl->newLazyProxy(static function () use ($mapper, $value, $objectMap, $mapTo) {
                    return $objectMap[$value] = $mapper->map($value, $mapTo->target);
                });
            }
```